rpc/handler.go

```diff
@@ -519,6 +519,27 @@ func (h *handler) runMethod(ctx context.Context, msg *jsonrpcMessage, callb *cal
 	return nil
 }
 
+var nullAsBytes = []byte{110, 117, 108, 108}
+
+// there are many avenues that could lead to an error being handled in runMethod, so we need to check
+// if nil has already been written to the stream before writing it again here
+func writeNilIfNotPresent(stream *jsoniter.Stream) {
+	b := stream.Buffer()
```

Comment:
1. if len of b < 4 then hasNil is true.  Is it bug (or i miss-understand)? 
2. It must work in most cases, but what if buffer already was flushed to client. Unlikely - in case of error - json is small. Maybe it’s fine.

Reply:
1. good catch!  I've added an else there now to set it to false.  If we've not got at least 4 bytes we can't possibly have written `null`.
2. I think in the case of a flush already having happened we've got more serious problems as we can't backtrack on that to the client.  Probably need to deal with those in a case by case basis if we ever found one.

Reply:
@hexoscott do we need check “stream” variable for nil?

Reply:
In this case we've already used `stream` before this call so can safely assume it won't be.  But nothing stopping other code calling this where it could be nil. I'll add the check into another PR shortly.